tools/migration/blacklist_migrator.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+from migrator_base import MigratorBase, get_arguments
+
+
+def main():
+    """ Starts the migration of blacklisted findings """
+    BlacklistMigrator(*get_arguments()).migrate()
+
+
+class BlacklistMigrator(MigratorBase):
+    """ Class for migrating a blacklist between two instances.
+    If some blacklisted finding cannot be found in the new instance, they will not
+    be migrated.
+    """
+    def migrate(self):
+        """ Migrates the blacklist. """
+        blacklist_infos = self.get_blacklist_infos()
+        if len(blacklist_infos) == 0:
+            self.logger.info("No new blacklisted findings to migrate")
+            exit(1)
+
+        self.logger.info("Migrating %s blacklisted findings" % len(blacklist_infos))
+        for blacklist_info in blacklist_infos:
+            old_id = blacklist_info["findingId"]
+            new_id = self.get_matching_finding_id(old_id)
+            if new_id is None:
+                self.logger.warning("Could not match finding %s to new instance" % self.get_findings_url(old_id))
+            else:
+                self.logger.info("Migrating blacklisted finding %s" % self.get_findings_url(old_id))
+                self.blacklist_finding(blacklist_info, new_id)
```

Comment:
check if already blacklisted?

Reply:
I removed that explicitly.
Often I was encountering the following problem.
Looking at the UI, a project had, for example, 18 blacklisted findings. Upon migrating the blacklist the result might have said something like: 6/34 findings where successfully blacklisted.
Now I wasn't sure ift there was actually a problem or not. When I removed the filtering of already blacklisted findings it said: 18/34 findings where blacklisted. Upon closer inspection I found out, that the other 16 findings did no longer exist on any project there.
So I kept it, because if the numbers where actually similar it is easier to determine if the migration was successful or not. We are not really running into any performance bottleneck here, so in the end it doesn't really matter.

Reply:
makes sense